### PR TITLE
Add connection report progress infrastructure

### PR DESCRIPTION
**Pull request by jramsay**
Fixes internal #1254

Add connection report progress infrastructure for TS

Note: support in other languages to follow

### Changes proposed: 
- update to consume latest ssh ssh-tcp packages that include reportProgress
- add reportProgress event 
- add connnection events

### Other Tasks:
- [ ] If you updated the Go SDK did you update the PackageVersion in tunnels.go

### CODE REVIEW

File: ts/src/connections/tunnelRelayTunnelHost.ts [hunk 1]
```diff
@@ -462,6 +463,7 @@ export class TunnelRelayTunnelHost extends TunnelConnectionSession implements Tu
             config.addService(PortForwardingService);
         });
         session.trace = this.trace;
+        session.onReportProgress(this.raiseReportProgress, this, this.sshSessionDisposables);
```
Comment (IlyaBiryukov) [inline ts/src/connections/tunnelRelayTunnelHost.ts:466]: Is this OK if progress is reported from multiple SSH sessions? The main one that connects sdk host to Relay, and different client sessions? The progress sequence may be quite chaotic.

Comment (jramsay) [reply ts/src/connections/tunnelRelayTunnelHost.ts:466]: I think so. The progress events aren't very verbose. Really only tracks 6 steps to help diagnose connection issues

Comment (IlyaBiryukov) [reply ts/src/connections/tunnelRelayTunnelHost.ts:466]: But there will be repetitive and out-of-order steps coming from different sessions.

Comment (jramsay) [reply ts/src/connections/tunnelRelayTunnelHost.ts:466]: thanks for pointing this case out! As discussed, I'll switch to using event args and add a clientSessionId that can be used to determine which client session is reporting the event.

**Review by IlyaBiryukov — APPROVED** [review]

**Pull request merged**